examples/receipt.py

```diff
@@ -0,0 +1,143 @@
+"""
+
+# Font "b" on my TM-P80-clone printer can squeeze 64 characters per line
+disclaimer_width = 64
+
+recipt_barcode = "1234567890"
+
+
+# Justify-text on left AND right sides by padding spaces,
+# code by: Georgina Skibinski https://stackoverflow.com/a/66087666
+def justify(txt: str, width: int) -> str:
+    prev_txt = txt
+    while (l := width - len(txt)) > 0:
+        txt = re.sub(r"(\s+)", r"\1 ", txt, count=l)
+        if txt == prev_txt:
+            break
+    return txt.rjust(width)
+
+
+p = printer.Usb(0x04B8, 0x0E20, profile="TM-P80")
+
+# Store Logo at the Top
+p.set_with_default()
+p.image("hocus-pocus.gif", center=True)
+
+# Print Address, centered
+p.ln(1)
+p.set_with_default(align="center")
+for l in address.split("\n"):
+    p.textln(l)
+p.ln(1)
+
+
+# Print date and time
+p.set_with_default(align="left")
+p.textln(timestamp.strftime("%A, %B %d, %Y   %I:%M%P"))
+
+# Print cashier's name
+p.set_with_default(align="right")
+p.textln("Served by: " + served_by)
+
+# Add some empty space before itemized list
+p.ln(2)
+
+
+## Add a bit of line spacing for itemized list for easier reading
+p.set_with_default()
+p.line_spacing(80, 180)
+
+## Itemized list header (bold with underline)
+p.set_with_default(bold=True, underline=True)
+p.textln(header_format.format(**header))
+
+## Itemized List
+p.set_with_default()
+for idx, item in enumerate(items):
+    txt = item_format.format(**item)
+    if idx == len(items) - 1:
+        # If this is the last item, add underline
+        # to visually "close" the list.
+        p.set_with_default(underline=True)
+    p.textln(txt)
+
+p.set_with_default()
+
+## Subtotal
+subtotal = sum([x["price"] for x in items])
+p.textln(subtotals_format.format(dummy="", desc="subtotal", price=subtotal))
+
+## Tax
+tax_amount = subtotal * tax_percent
+tax_desc = "Guild Tax (%d%%)" % (int(tax_percent * 100.0))
+p.textln(subtotals_format.format(dummy="", desc=tax_desc, price=tax_amount))
+
+## Total
+## NOTE: because we use double-sized font, alignment won't match
+## the previous lines. Instead, with trim leading whitespace,
+## and use the printer's built-in right-alignment feature.
+p.set_with_default(align="right", custom_size=True, width=2, height=2)
+total_amount = subtotal + tax_amount
+total_desc = "Total"
+total_text_line = subtotals_format.format(dummy="", desc=total_desc, price=total_amount)
+total_text_line = total_text_line.strip()
+p.textln(total_text_line)
+
+# Add some empty space before disclaimer
+p.ln(4)
+
+# preprocess disclaimer text:
+# In python it's easy to use multilined strings,
+# but to print the disclaimer we want to merge lines
+# and condense whitespaces.
+txt = disclaimer.replace("\n", " ")
+txt = re.sub("  +", " ", txt)
+# textwrap.wrap() ensures words are not broken (unlike "escpos.block_text()").
+txt = textwrap.wrap(txt, width=disclaimer_width)
+# Justify each line
+txt = [justify(x, disclaimer_width) for x in txt]
+p.set_with_default(font="b")
+for l in txt:
+    p.textln(l)
```

Comment:
```suggestion
p.software_columns(text_list=[txt], widths=disclaimer_width, align='left')
```
Again, the [software_columns](https://github.com/python-escpos/python-escpos/blob/99501cc2c1d21ea156b8b8b530ad66c9c9d00fe3/src/escpos/escpos.py#L976) method is your friend here.
This will print the text in `disclaimer` as a single column of left aligned text wrapped to fit within 64 characters max width per line.
The only lack would be the justification of the text that I would like to see be part of the python-escpos tools in the near future.